### PR TITLE
Add CI Workflow for Examples

### LINKED ISSUE #2: [CI] Add workflow to build and publish examples
Since the typescript examples have been moved from the rust host monorepo into this language repo, we need to migrate the pipelines that build and publish the examples as OCI images in the ghcr.io registry.

### Information
- Examples are under the [/examples](https://github.com/wasmCloud/typescript/tree/main/examples)
- Existing Pipelines for building and publishing examples can be used as a reference from the [rust host monorepo](https://github.com/wasmCloud/wasmCloud/blob/24789dff476f3ed3a64faefa2222d8ea4fe77bdd/.github/workflows/examples-components.yml#L235-L248)

### PR DESCRIPTION
**Pull request by lachieh**
## Feature or Problem
Adds examples workflow for building, testing and releasing the typescript component examples.

## Related Issues
Closes #2

### CODE REVIEW

File: examples/components/http-password-checker/src/http-password-checker.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ export const incomingHandler = {
         // Perform the check
         try {
           const result = await handlePasswordCheck(cr);
-          return await sendResponseJSON(resp, 200, Response.ok(result));
+          return await sendResponseJSON(resp, 200, result);
```
Comment (lachieh) [inline examples/components/http-password-checker/src/http-password-checker.ts:55]: Small bug, this was causing nested success objects like so:
```jsonc
{
  "status": "success",
  "data": {
    "status": "success",
    "data": {
      // ...
    }
  }
}
```

**Review by lachieh — COMMENTED** [review]
@brooksmtownsend can you take a look at this one? TIA!

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 ## GitHub CI
 
-.github @wasmCloud/ci-maintainers
+.github/** @wasmCloud/ci-maintainers
 .github/CODEOWNERS @wasmCloud/org-maintainers
```
[the remark this answers is not in the document]
Comment (lachieh) [reply .github/CODEOWNERS:10]: Nevermind, I found the settings for this repo.

**Comment by joonas** [comment]
Just in case, you can use the latest version of [`taiki-e/install-action`](https://github.com/taiki-e/install-action/releases/tag/v2.47.0) to handle cached installation of `wash`

**Comment by lachieh** [comment]
Ooh, yes, I will update the action!

**Comment by lachieh** [comment]
@joonas I made that change! I also added you as a reviewer if you want to provide any feedback.

File: .github/workflows/examples_.yml
Comment (brooksmtownsend) [inline .github/workflows/examples_.yml:1]: nit: accidental `_` at end of filename?

File: .github/workflows/examples_.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+              - .github/workflows/examples_component.yml
+              - .github/workflows/examples_provider.yml
+            components:
+              - examples/components/**
+            providers:
+              - examples/providers/**
+
+      - name: Install `wash` CLI
+        uses: taiki-e/install-action@acd25891978b4cdaebd139d3efef606d26513b14 # v2.47.0
+        with: { tool: wash }
```
Comment (brooksmtownsend) [inline .github/workflows/examples_.yml:47]: 😲

**Review by brooksmtownsend — APPROVED** [review]
Lots of cool stuff here 😅 Left one nit

File: .github/workflows/examples_.yml
Comment (lachieh) [reply .github/workflows/examples_.yml:1]: Screenshot 2024-12-30 at 3 28 50 PM

**Pull request merged**